src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2196,7 +2196,7 @@ private void ExpandByABlock(int minBlockCharCount)
             // - We'd also prefer to make it at least at big as the current length (thus doubling capacity).
             //   - But this is only up to a maximum, so we stay in the small object heap, and never allocate
             //     really big chunks even if the string gets really big.
-            int newBlockLength = Math.Max(minBlockCharCount, Math.Min(Length, MaxChunkSize));
+            int newBlockLength = Math.Clamp(Length, minBlockCharCount, MaxChunkSize);
```

Comment:
This will now throw if MaxChunkSize < minBlockCharCount. I don't think it can be, though.